sunpy/coordinates/wcs_utils.py

```diff
@@ -41,11 +41,31 @@ def solar_wcs_frame_mapping(wcs):
     required_attrs = {HeliographicStonyhurst: ['hgln_obs', 'hglt_obs', 'dsun_obs'],
                       HeliographicCarrington: ['crln_obs', 'hglt_obs', 'dsun_obs']}
 
+    # Get rsun from the WCS auxillary information
+    rsun = wcs.wcs.aux.rsun_ref
+    if rsun is not None:
+        rsun *= u.m
+
+    # This custom attribute was always used in sunpy < 2.1; these warnings
+    # can be converted into errors in sunpy 3.1
+    if hasattr(wcs, 'rsun'):
+        warnings.warn('Support for the .rsun attribute on a WCS is deprecated. '
+                      'Set observer keywords in the FITS header, or directly set the wcs.wcs.aux '
+                      'values instead.',
+                      SunpyUserWarning)
+        if rsun is None:
+            rsun = wcs.rsun
+        else:
+            warnings.warn('rsun information present in WCS auxillary information, ignoring '
+                          '.rsun')
```

Comment:
This should be a sunpyuserwarning.

Reply:
Since this would need to be backported, I created #5219